# Confidence Interval of Polling Figures



You are running a political campaign and decide to run 30 focus groups with about 10 people in each group. You get the results and want to report to your candidate the number of people who would vote for them in a typical 10-person group. Since there is some variability in each focus group, you decide that the most accurate way is to give a 95% z-confidence interval. You assume from past experience that the standard deviation is 2.89.

1.Import the random Python package and set the seed to 39809. This will ensure that we get the same results every time we run the program:


In [1]:
import numpy as np
from scipy import stats
import random
import math
import matplotlib.pyplot as plt
random.seed(39809)

2.Initialize our sample list and collect our samples from our focus groups. Use random.randint

In [2]:
sample_list = []

for i in range(30):
    sample_list.append(random.randint(0, 10))
random.seed(39809)

In [3]:
sample_mean = np.mean(sample_list)
sample_mean

5.0

3.Calculate 95% z-confidence interval.

In [4]:
n = len(sample_list)
cl = 0.95
std = 2.89

In [5]:
critic_value = stats.norm.ppf(((1-cl)/2) + cl)
critic_value

1.959963984540054

In [6]:
lower_limit = sample_mean - (critic_value * (std/math.sqrt(n)))
lower_limit

3.965845784931483

In [7]:
upper_limit = sample_mean + (critic_value * (std/math.sqrt(n)))
upper_limit

6.034154215068517

4.If you did everything correctly, then the following should be printed when you run your notebook:

Your 0.95 z confidence interval is (3.965845784931483, 6.034154215068517)

In [8]:
print(f'Your {cl} z confidence interval is ({lower_limit:.2f}, {upper_limit:.2f})')

Your 0.95 z confidence interval is (3.97, 6.03)


# Hypothesis Testing



Your boss asks you to conduct a hypothesis test about the mean dwell time of a new type of UAV. Before you arrived, an experiment was conducted on n=5 UAVs (all of the new type) resulting in a sample mean dwell time of hours. The goal is to conclusively demonstrate, if possible, that the data supports the manufacturer’s claim that the mean dwell time is greater than 10 hours. Given that it is reasonable to assume the dwell times are normally distributed, the sample standard deviation is s = 0.5 hours, and using a significance level of α = 0.01:

1.Write out the null and alternative hypotheses


H0: t_dwell > 10
H1: t_dwell <=10

2.Calculate the test statistic

In [11]:
import pandas as pd
import scipy.stats as stats
import math

assume that the mean time is 9.5 hours (which is not given)

In [14]:
sample_list2 = []

for i in range(5):
    sample_list2.append(random.randint(8, 12))
random.seed(39809)

In [18]:
onesample = stats.ttest_1samp(sample_list2, 10)

In [19]:
onesample.statistic

0.3429971702850165

3.Find the p-value and state the outcome

In [20]:
onesample.pvalue

0.7488684500235274

Probably there is a mistake here. 5 values are two low - even for a t test.